tzng parser panics on malformed record

```sh
10:45 ~/work/brim/zq(master)
$ git rev-parse --short HEAD
6a05953
10:45 ~/work/brim/zq(master)
$ cat blar.zng
#0:record[blar:string]
0:[blar1]
10:45 ~/work/brim/zq(master)
$ zq blar.zng
panic: runtime error: slice bounds out of range [7:6]
```

--- a/tests/suite/errors/nosemi.yaml
+++ b/tests/suite/errors/nosemi.yaml
@@ -1,12 +1,7 @@
-script: zq in.tzng
+zql: filter *
 
-inputs:
-  - name: in.tzng
-    data: |
-        #0:record[a:string]
-        0:[1]
+input: |
+  #0:record[x:int32]
+  0:[1]
 
-outputs:
-  - name: stderr
-    regexp: |
-      unterminated container
+errorRE: unterminated container

--- a/tests/suite/errors/nosemi.yaml
+++ b/tests/suite/errors/nosemi.yaml
@@ -1,0 +1,12 @@
+script: zq in.tzng
+
+inputs:
+  - name: in.tzng
+    data: |
+        #0:record[a:string]
+        0:[1]
+
+outputs:
+  - name: stderr
+    regexp: |
+      unterminated container

--- a/zio/tzngio/parser.go
+++ b/zio/tzngio/parser.go
@@ -135,6 +135,9 @@ func zngParseField(builder *zcode.Builder, typ zng.Type, b []byte) ([]byte, erro
 			i++
 		}
 	}
+	if i == len(b) {
+		return nil, ErrUnterminated
+	}
 
 	zv, err := realType.Parse(b[:i])
 	if err != nil {

--- a/ztest/ztest.go
+++ b/ztest/ztest.go
@@ -582,7 +582,7 @@ func runzq(bindir, ZQL, outputFormat, outputFlags string, inputs ...string) (out
 	zctx := resolver.NewContext()
 	zr, err := loadInputs(inputs, zctx)
 	if err != nil {
-		return "", "", err
+		return "", err.Error(), err
 	}
 	defer zr.Close()
 	if outputFormat == "types" {

